### PR TITLE
chore: install secret detection on pre-commit hooks

### LINKED ISSUE #2290: chore: install secret detection on pre-commit hooks
to catch secret before it leaves your machine and ease remediation.

### Description
As a developer, I want to incorporate a check so that any secrets/crypto doesn't get pushed to the github repo and this check runs as a part of pre-commit check

### Acceptance Criteria
1. Test should be done by incorporating various credentials/secrets like github secrets, aws secrets, pem files, access tokens

### PR DESCRIPTION
**Pull request by zondervancalvez**
## Commit to be reviewed

chore: install secret detection on pre-commit hooks
```
Primary Changes
----------------
1. Installed gitleaks for secret detection.
2. This pre-commit checker detects any secrets
or crypto so that it doesn't get pushed to
the github repo.
3. Added script to run install and uninstall
the pre-commit hooks in package.json

```
Fixes #2290

**Pull Request Requirements**
- [ ] Rebased onto `upstream/main` branch and squashed into single commit to help maintainers review it more efficient and to avoid spaghetti git commit graphs that obfuscate which commit did exactly what change, when and, why.
- [ ] Have git sign off at the end of commit message to avoid being marked red. You can add `-s` flag when using `git commit` command. You may refer to this [link](https://docs.github.com/en/authentication/managing-commit-signature-verification/signing-commits) for more information.
- [ ] Follow the Commit Linting specification. You may refer to this [link](https://www.conventionalcommits.org/en/v1.0.0-beta.4/#specification) for more information. 

**Character Limit**
- [ ] Pull Request Title and Commit Subject must not exceed 72 characters (including spaces and special characters).
- [ ] Commit Message per line must not exceed 80 characters (including spaces and special characters).

**A Must Read for Beginners**
For rebasing and squashing, here's a [must read guide](https://github.com/servo/servo/wiki/Beginner's-guide-to-rebasing-and-squashing) for beginners.

### CODE REVIEW

**Comment by zondervancalvez** [comment]
image

**Review by petermetz — CHANGES_REQUESTED** [review]
@zondervancalvez A good start, but please add an npm script in the top level package.json that coldstarts the pre-commit hook. The idea is that on-boarding new contributors could run that script ones and have the commit hook enabled after it finished running.

Please also add another npm script that removes the pre-commit hook and cleans up after itself (files downloaded/created during the enabler script )

You could call the scripts `tools:install-pre-commit-secret-detection` and `tools:uninstall-pre-commit-secret-detection`

**Comment by petermetz** [comment]
@zondervancalvez Are you still working on this?

**Comment by zondervancalvez** [comment]
> @zondervancalvez Are you still working on this?

@petermetz Yes

**Review by petermetz — APPROVED** [review]

**Comment by jagpreetsinghsasan** [comment]
@hyperledger-cacti/cacti-maintainers requesting review on this PR

File: package.json [hunk 1]
```diff
@@ -33,6 +33,8 @@
     "set-yarn-version": "yarn set version stable",
     "enable-corepack": "npm i -g corepack && corepack enable && corepack prepare yarn@4.3.1 --activate",
     "custom-checks": "TS_NODE_PROJECT=./tools/tsconfig.json node --trace-deprecation --experimental-modules --abort-on-uncaught-exception --loader ts-node/esm --experimental-specifier-resolution=node ./tools/custom-checks/run-custom-checks.ts",
+    "tools:install-pre-commit-secret-detection": "pre-commit install && pre-commit autoupdate",
+    "tools:uninstall-pre-commit-secret-detection": "pre-commit uninstall",
```
Comment (outSH) [inline package.json:37]: Could you also update our `CONTRIBUTING.md` and / or `BUILD.md` to recommend installing these pre-commits when setting up the environment?

**Review by outSH — CHANGES_REQUESTED** [review]

**Review by outSH — APPROVED** [review]
LGTM, thank you!

**Pull request merged**